### PR TITLE
Graylog Provider

### PR DESCRIPTION
**Pull request by jahunt1**
- Changed attribute accessor in AnalyticsEvent to return Map instead of HashMap
 - new provider for Graylog HTTP inputs
 - some Javadoc updates

### CODE REVIEW

File: graylog-provider/src/main/java/com/busybusy/graylog_provider/EventJsonizer.java [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+                else if (attributeValue instanceof List)
+                {
+                    sb.append(jsonAttribute).append("\": ").append(getJsonFromListRecursive((List) attributeValue));
+                }
+                else if (attributeValue instanceof Map)
+                {
+                    //noinspection unchecked
+                    sb.append(jsonAttribute).append("\": ").append(getJsonFromMapRecursive((Map<String, Object>) attributeValue));
+                }
+                else if (attributeValue instanceof Integer ||
```
Comment (simon-the-canadian) [inline graylog-provider/src/main/java/com/busybusy/graylog_provider/EventJsonizer.java]: seems like you could have an isNumber() method that would return a boolean and execute this OR statement and use it in the 3 spots that do this check

File: graylog-provider/src/test/java/com/busybusy/graylog_provider/GraylogProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+                .setTimed(true)
+                .putAttribute("some_param", "yes")
+                .putAttribute("another_param", "yes again");
+
+        provider.sendEvent(event);
+        assertFalse(logEventCalled);
+
+        try
+        {
+            Thread.sleep(50);
```
Comment (simon-the-canadian) [inline graylog-provider/src/test/java/com/busybusy/graylog_provider/GraylogProviderTest.java]: is there a need to sleep here if you're having the lock wait on the next couple of lines?

Comment (jahunt1) [inline graylog-provider/src/test/java/com/busybusy/graylog_provider/GraylogProviderTest.java]: Nope. At one point I was validating that we get a positive duration on the timed event, but now that the event gets serialized I don't care about parsing the JSON I would get out of the callback.

File: graylog-provider/src/main/java/com/busybusy/graylog_provider/EventJsonizer.java (hunk 1 above)
Comment (jahunt1) [inline graylog-provider/src/main/java/com/busybusy/graylog_provider/EventJsonizer.java]: Such better. Very wisdom.

**Pull request merged**